core/services/workflows/syncer/workflow_registry.go

```diff
@@ -362,6 +355,192 @@ func (w *workflowRegistry) readRegistryEvents(ctx context.Context, don capabilit
+				continue
+			}
+
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, the metadata is the same, and the status is active
+		// then the workflow is unchanged. mark as seen.
+		if wfMeta.Status == 1 {
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, the metadata is the same, but the status isn't active
+		// then the workflow is has been paused. NOOP to be handled below as a deleted event, which clears the DB workflow spec.
+	}
+
+	// Shut down engines that are no longer in the contract's latest workflow metadata state
+	allEngines := w.engineRegistry.GetAll()
+	for _, engine := range allEngines {
+		keyStr := engine.workflowName + hex.EncodeToString(engine.workflowOwner)
+		_, exists := engineKeyMap[keyStr]
+
+		if !exists {
+			toDeletedEvent := WorkflowRegistryWorkflowDeletedV1{
+				WorkflowID:    engine.workflowID,
+				WorkflowOwner: engine.workflowOwner,
+				DonID:         donID,
+				WorkflowName:  engine.workflowName,
+			}
+			err := w.handler.Handle(ctx, workflowAsEvent{
+				Data:      toDeletedEvent,
+				EventType: WorkflowDeletedEvent,
+			})
+			if err != nil {
+				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowDeletedEvent)
+			}
+		}
+	}
+}
+
+// pollRegistryState polls every w.ticker tick for the latest workflow state and
+// calls the handler to bring the system's state up to date with the difference.
+func (w *workflowRegistry) pollRegistryStateToEvents(ctx context.Context, don capabilities.DON, reader ContractReader) {
+	ticker := w.getTicker()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker:
+			workflowMetadata, _, err := w.getWorkflowMetadata(ctx, don, reader)
+			if err != nil {
+				w.lggr.Errorw("failed to process registry state", "err", err)
+				continue
+			}
+			w.handleWorkflowMetadataAsEvents(ctx, workflowMetadata, don.ID)
+		}
+	}
+}
+
+// syncUsingReconciliationStrategy syncs workflow registry contract state by polling the workflow metadata state and comparing to local state.
+// It still watches for ForceUpdateSecretsEvents, which can't be reconciled through workflow metadata state.
+func (w *workflowRegistry) syncUsingReconciliationStrategy(ctx context.Context, don capabilities.DON, reader ContractReader) {
+	workflowMetadata, loadWorkflowsHead, err := w.getWorkflowMetadata(ctx, don, reader)
+	// TODO: if this fails, we should still start the event looping
+	if err != nil {
+		w.lggr.Errorw("failed to load initial workflows", "err", err)
+		return
+	}
+	w.handleWorkflowMetadataAsEvents(ctx, workflowMetadata, don.ID)
+
+	w.pollRegistryStateToEvents(ctx, don, reader)
+
+	// Event polling for only ForceUpdateSecretsEvent
+	ets := []WorkflowRegistryEventType{
+		ForceUpdateSecretsEvent,
+	}
+
+	w.readRegistryEvents(ctx, ets, don, reader, loadWorkflowsHead.Height)
```

Comment:
I don't think this is doing what you intended -- poll is polling in a for/select but so is readRegistsryEvents

i.e. we will never get to readRegistryEvents

Another reason to use some goroutines plus a channel to send events to 😉

Reply:
Right. Hadn't got to that yet. Take another look now.